### ISSUE TITLE
FASTA truncates long paths

### ISSUE THREAD

**Issue by shelkmike** [wrpearson/fasta36#26]
I run the command
`/home/shelkmike/Work/Tools/FASTA/fasta-36.3.8h/bin/glsearch36 -b 1 -d 1 -H -m 8C -n -O ./Plants_group_number_1/alignment_between_the_close_photosynthetic_species_and_the_distant_photosynthetic_species_abcdabcdabcd.txt -T 20 ./Plants_group_number_1/rps19_of_the_close_photosynthetic_species.fasta ./Plants_group_number_1/distant_photosynthetic_species_plastid_genome.fasta`

It should produce a file named alignment_between_the_close_photosynthetic_species_and_the_distant_photosynthetic_species_abcdabcdabcd.txt

However, the created file is named alignment_between_the_close_photosynthetic_species_and_the_distant_photosynthetic_species_abcda 

Basically, FASTA truncates output paths that are too long. 
Could you fix this please?

Best wishes

**Comment by wrpearson** [comment]
I guess I am not surprised that the "-O" option does not allow very long file names.  A more reliable way to send your output to a file is using the ">" (stdout) symbol.



/home/shelkmike/Work/Tools/FASTA/fasta-36.3.8h/bin/glsearch36 -b 1 -d 1 -H -m 8C -n -T 20 ./Plants_group_number_1/rps19_of_the_close_photosynthetic_species.fasta ./Plants_group_number_1/distant_photosynthetic_species_plastid_genome.fasta > ./Plants_group_number_1/alignment_between_the_close_photosynthetic_species_and_the_distant_photosynthetic_species_abcdabcdabcd.txt

Begin forwarded message:

**Comment by shelkmike** [comment]
Thank you, writing to a file through the standard output solves the problem. There is also another problem with long paths.

If I run
`/home/shelkmike/Work/Tools/FASTA/fasta-36.3.8h/bin/glsearch36 -b 1 -d 1 -H -m 8C -n -T 22 ./Plants_group_number_1/ycf1_of_the_close_photosynthetic_species.fasta ./Plants_group_number_1/nonphotosynthetic_species_plastid_genome.fasta >./Plants_group_number_1/ycf2_alignment_between_the_close_photosynthetic_species_and_the_nonphotosynthetic_species.txt`

the program works correctly, but if I indicate full paths like

`/home/shelkmike/Work/Tools/FASTA/fasta-36.3.8h/bin/glsearch36 -b 1 -d 1 -H -m 8C -n -T 22 /home/shelkmike/Work/While_Makarich_doesnt_work/Analyse_mutations_with_respect_to_the_last_common_ancestor/Plants_group_number_1/ycf1_of_the_close_photosynthetic_species.fasta /home/shelkmike/Work/While_Makarich_doesnt_work/Analyse_mutations_with_respect_to_the_last_common_ancestor/Plants_group_number_1/nonphotosynthetic_species_plastid_genome.fasta >/home/shelkmike/Work/While_Makarich_doesnt_work/Analyse_mutations_with_respect_to_the_last_common_ancestor/Plants_group_number_1/ycf2_alignment_between_the_close_photosynthetic_species_and_the_nonphotosynthetic_species.txt`

the program writes

> ERROR [comp_lib9.c:642] cannot open library /home/shelkmike/Work/While_Makarich_doesnt_work/Analyse_mutations_with_respect_to_the_last_common_ancestor/Plants_group_x

The difference between the first and the second commands is that in the first I use relative paths ("./") and in the second I use full paths ("/home/shelkmike/...").

Using relative paths is not a big problem for me but this is nevertheless a bug.

**Comment by wrpearson** [comment]
The maximum length of a file name is set in src/defs.h at MAX_FN 120.

Increasing that constant and re-compiling would allow longer file names.

Bill Pearson

Begin forwarded message:
